Dockerfile

```diff
@@ -0,0 +1,47 @@
+FROM ubuntu:22.04
```

Comment:
Why do we need a dockerfile? I rather have it building on a plain github action machine.. I'm trying to get to the point we can do this also for eclair.

Reply:
The dockerfile is not intended for adding MacOS support in CI. It is intended for people like me who currently don't have access to a Linux machine, because their primary development machine runs on MacOS. Without the dockerfile my development workflow would be hampered, so I am hoping we can agree on keeping it.

Reply:
What's the problem in macos you're having? That llvm@17 isn't available there? LLVM 15 or 16 also work with llvm-codegen (I think atleast 15 should be available on macos).

Reply:
@luc-tielen I integrated your changes and removed the Dockerfile.

Build is passing for me locally on MacOS also with the following steps
```bash
$ brew install llvm@16
$ PATH="/usr/local/opt/llvm/bin:$PATH" LDFLAGS="-L/usr/local/opt/llvm/lib" CPPFLAGS="-I/usr/local/opt/llvm/include" make test
```

Reply:
Yeah LLVM needs to be in the $PATH. Weird that brew doesn't do it automatically..